typescript/lib/services/service_context.ts

```diff
@@ -0,0 +1,125 @@
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getServiceID(): ServiceID { 
+        return this.serviceId;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getIPAddress(): string {
+        return this.ipAddress;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public execCommand(command: string[]): [number, [], Error] { //TODO - [int32, pointer to []byte (comment), error ] ; is is not []`type` like []string
+        let serviceId = this.serviceId;
+        let args = constructors.newGetExecCommandArgs(serviceId, command);
+        let request: grpc.requestCallback<bindingsJs.ExecCommandResponse>;
+
+        let resp, err = this.client.execCommand(args, request); //TODO - does error checking work here ? 
+        
+        if (err != null) {
+            return [0, null, new Error( //TODO - Error type (do I need callback?)
+			    "An error occurred executing command '%v' on service '%v'", //TODO - I might not be able keep the '%v'
+            )]; //TODO (comment) cannot return multipe values in typescript function, need to wrap in array or object
+        }
+        return resp.ExitCode, resp.LogOutput, null; //TODO (comment) resp.LogOuput was a pointer
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public generateFiles(filesToGenerateSet: Set<string>): [Map<string, GeneratedFileFilepaths>, Error] {
+        let serviceId = this.serviceId;
+        let fileGenerationOpts = new Map(); //TODO - Still need map since both key and value are useful ; Key (Set just needs key): string & Value: core_api_bindings.FileGenerationOptions{} - type not needed in typescript
+        for (let fileId in filesToGenerateSet) {
+            fileGenerationOpts[fileId] = new bindingsJs.FileGenerationOptions(); //TODO (comment) this was a pointer
+            fileGenerationOpts[fileId].setFileTypeToGenerate(bindingsJs.FileGenerationOptions.FileTypeToGenerate.FILE);
+        }
+
+        let args = constructors.newGetGenerateFilesArgs(serviceId, fileGenerationOpts);
+        let request: grpc.requestCallback<bindingsJs.GenerateFilesResponse>;
+
+        let resp, err = this.client.generateFiles(args, request);
+        if (err != null){
+            return [null, new Error("An error occurred generating files using args: %+v")]; //TODO - can't use console.error ; no %+v maybe
+        }
+        let generatedFileRelativeFilepaths = resp.GeneratedFileRelativeFilepaths;
+
+        let result = new Map(); // TODO Key:string & Value:GeneratedFileFilepaths{} - type not needed in typescript
+        for (let fileId in filesToGenerateSet) {
+            let relativeFilepath, found = generatedFileRelativeFilepaths[fileId]; //TODO - does this work (having map values return 2 things??)
```

Comment:
Oh, I would need to fix like how I did in the `loadStaticFiles()` method - line 118